nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -6233,27 +6237,54 @@ protected Collection<AbstractMetricsRegistry> populateFlowMetrics() {
+
+            final ForkJoinPool.ForkJoinWorkerThreadFactory factory = pool -> {
+                final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
+                worker.setName("analytics-prediction-parallel-processing-thread-" + UUID.randomUUID());
+                return worker;
+            };
+
+            final ForkJoinPool parallelProcessingThreadPool = new ForkJoinPool(parallelProcessingThreads, factory, null, false);
             try {
-                final StatusAnalytics statusAnalytics = controllerFacade.getConnectionStatusAnalytics(c.getIdentifier());
-                PrometheusMetricsUtil.createConnectionStatusAnalyticsMetrics(connectionAnalyticsMetricsRegistry,
-                        statusAnalytics,
-                        instanceId,
-                        "Connection",
-                        c.getName(),
-                        c.getIdentifier(),
-                        c.getProcessGroup().getIdentifier(),
-                        c.getSource().getName(),
-                        c.getSource().getIdentifier(),
-                        c.getDestination().getName(),
-                        c.getDestination().getIdentifier()
-                );
-                PrometheusMetricsUtil.aggregateConnectionPredictionMetrics(aggregatedMetrics, statusAnalytics.getPredictions());
-            } catch (ResourceNotFoundException rnfe) {
-                break;
+                parallelProcessingThreadPool.submit(
+                        () -> connections.parallelStream().forEach((c) -> {
+                            final StatusAnalytics statusAnalytics = controllerFacade.getConnectionStatusAnalytics(c.getIdentifier());
+                            PrometheusMetricsUtil.createConnectionStatusAnalyticsMetrics(connectionAnalyticsMetricsRegistry,
+                                    statusAnalytics,
+                                    instanceId,
+                                    "Connection",
+                                    c.getName(),
+                                    c.getIdentifier(),
+                                    c.getProcessGroup().getIdentifier(),
+                                    c.getSource().getName(),
+                                    c.getSource().getIdentifier(),
+                                    c.getDestination().getName(),
+                                    c.getDestination().getIdentifier()
+                            );
+                            predictions.add(statusAnalytics.getPredictions());
+                        }));
+            } finally {
+                parallelProcessingThreadPool.shutdown();
+                try {
+                    boolean finished = parallelProcessingThreadPool.awaitTermination(parallelProcessingTimeout, TimeUnit.MILLISECONDS);
+                    if (!finished) {
```

Comment:
If it doesn't finish in time, the result is irrelevant so we should use `parallelProcessingThreadPool.shutdownNow()` in this instance also.